package/MDAnalysis/analysis/distances.py

```diff
@@ -156,7 +163,7 @@ def dist(A, B, offset=0, box=None):
     if A.atoms.n_atoms != B.atoms.n_atoms:
         raise ValueError("AtomGroups A and B do not have the same number of atoms")
     try:
-        off_A, off_B = offset
+        off_A, off_B = int(offset)
     except (TypeError, ValueError):
         off_A = off_B = int(offset)
```

Comment:
Mypy sees `offset` as possibly a tuple causing:

```
package/MDAnalysis/analysis/distances.py:168: error: Argument 1 to "int" has incompatible type "Union[int, Tuple[Any, ...]]"; expected "Union[str, bytes, array[Any], mmap, _CData, PickleBuffer, SupportsInt, SupportsIndex, SupportsTrunc]"
```

We may be able just to get rid of the type conversion.
```suggestion
        off_A = off_B = offset
```